test/util/test_runners/integration_test_case.py

```diff
@@ -0,0 +1,89 @@
+"""
+extend the unitest TestCase class
+"""
+
+# Builtins
+import io
+import sys
+
+# Third party
+import itertools
+from unittest import TestCase
+from contextlib import redirect_stdout
+import subprocess as sp
+from unittest.mock import patch, MagicMock
+
+# Internals
+from conjur.cli import Cli
+from test.util.test_runners.params import ClientParams, TestEnvironmentParams
+
+
+class IntegrationTestCaseBase(TestCase):
+
+    def __init__(self, testname, client_params: ClientParams = None,
+                 environment_params: TestEnvironmentParams = None):
+        """
+        Base class that extends unittest TestCase
+        used to setup tests environment
+        """
+        self.environment = TestEnvironmentParams() if environment_params is None else environment_params
+        self.client_params = ClientParams() if client_params is None else client_params
+
+        super(IntegrationTestCaseBase, self).__init__(testname)
+
+
+    def invoke_cli(self, *args, exit_code=0) -> str:
+        """
+        Invoker for the integration tests.
+        if self.environment.invoke_process == True
+        than it will call the cli binaries to run the integration tests
```

Comment:
then*

what do you mean here by CLI binaries?

Reply:
pyinstaller output (conjur)

Reply:
I would change it to CLI executable so we are speaking the same language in the repo